app/Main.hs

```diff
@@ -14,24 +15,38 @@ import Hie.Locate
 import Hie.Yaml
 import System.Directory
 import System.Directory.Internal
-import System.FilePath.Posix
+import System.FilePath
+import Hie.CabalHelper
+import Distribution.Helper (Ex (..))
 
 main :: IO ()
 main = do
   pwd <- getCurrentDirectory
-  files <- listDirectory pwd
-  let name =
-        if  | any (("dist-newstyle" ==) . takeFileName) files -> "cabal"
-            | any ((".stack-work" ==) . takeFileName) files -> "stack"
-            | any (("cabal.project" ==) . takeFileName) files -> "cabal"
-            | any (("stack.yaml" ==) . takeFileName) files -> "stack"
-            | otherwise -> "cabal"
-  cfs <- runMaybeT $ case name of
-    "cabal" -> cabalPkgs pwd
-    _ -> stackYamlPkgs pwd
-  when (null cfs) $ error $
-    "No .cabal files found under"
-      <> pwd
-      <> "\n You may need to run stack build."
-  pkgs <- catMaybes <$> mapM (nestedPkg pwd) (concat cfs)
-  putStr <$> hieYaml name $ fmtPkgs name pkgs
+  projM <- findCabalHelperEntryPoint $ pwd </> "Foo.hs"
```

Comment:
This is okay, but should be optional and behind the cabal-helper flag.

Reply:
True, it is a show-case, it should also not remove the old behaviour as it is waaaaaaaaayyy faster and should be the default mode. So, adding a cli would be sensible, imo, via optparse-applicative